torchdata/dataloader2/reading_service.py

```diff
@@ -259,8 +259,7 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
         replicable_dp = replicable_dps[0]
 
         if self.worker_prefetch_cnt > 0:
-            worker_prefetch_dp = replicable_dp.prefetch(self.worker_prefetch_cnt)
-            replicable_dp = worker_prefetch_dp
+            replicable_dp = replicable_dp.prefetch(self.worker_prefetch_cnt)
```

Comment:
Interesting `worker_prefetch_dp` wasn't used anywhere else

Reply:
It was my bad, I was testing the graph earlier in the PR of dispatching process but forgot to optimize this part.